vault/resource_terraform_cloud_secret_creds.go

```diff
@@ -0,0 +1,148 @@
+				Sensitive:   true,
+			},
+			"token_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the Terraform Token provided",
+			},
+			"organization": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				Description: "Name of the Terraform Cloud or Enterprise organization",
+			},
+			"team_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				Description: "ID of the Terraform Cloud or Enterprise team under organization (e.g., settings/teams/team-xxxxxxxxxxxxx)",
+			},
+		},
+	}
+}
+
+func createTerraformCloudSecretCredsResource(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	backend := d.Get("backend").(string)
+	role := d.Get("role").(string)
+	path := fmt.Sprintf("%s/creds/%s", backend, role)
+
+	secret, err := client.Logical().Read(path)
+	if err != nil {
+		return fmt.Errorf("error reading from Vault: %s", err)
+	}
+	log.Printf("[DEBUG] Read %q from Vault", path)
+
+	if secret == nil {
+		return fmt.Errorf("no role found at %q", path)
+	}
+
+	token := secret.Data["token"].(string)
+	if token == "" {
+		return fmt.Errorf("token is not set in response")
+	}
+	tokenId := secret.Data["token_id"].(string)
+
+	organization := secret.Data["organization"]
+	teamId := secret.Data["team_id"]
+
+	if secret.LeaseID != "" {
+		d.SetId(secret.LeaseID)
+	} else {
+		d.SetId(tokenId)
+	}
+	d.Set("token", token)
+	d.Set("token_id", tokenId)
+	d.Set("organization", organization)
+	d.Set("team_id", teamId)
+
+	return nil
+}
+
+func deleteTerraformCloudSecretCredsResource(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	id := d.Id()
+
+	err := client.Sys().Revoke(id)
+	if err != nil {
+		return fmt.Errorf("error revoking token from Vault: %s", err)
+	}
+	log.Printf("[DEBUG] Revoked tokenId: %q from Vault", id)
+	return nil
+}
+
+func updateTerraformCloudSecretCredsResource(d *schema.ResourceData, meta interface{}) error {
+	err := deleteTerraformCloudSecretCredsResource(d, meta)
+	if err != nil {
+		return fmt.Errorf("previous token not revoked: %s", err)
+	}
+	err = createTerraformCloudSecretCredsResource(d, meta)
+	return err
+}
+
+func readTerraformCloudSecretCredsResource(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	id := d.Id()
+	organization := d.Get("organization")
+	teamId := d.Get("team_id")
+
+	if organization == "" && teamId == "" {
+		data := map[string]interface{}{
+			"lease_id": id,
+		}
+		creds, err := client.Logical().Write("/sys/leases/lookup", data)
+		if err != nil {
+			if strings.Contains(err.Error(), "lease not found") {
+				return nil
+			} else {
+				return err
+			}
+		}
+		if creds.LeaseDuration <= 0 {
+			return nil
+		}
+
+		d.SetId(id)
+		d.Set("token", data["token"])
+		d.Set("token_id", data["token_id"])
+		d.Set("organization", data["organization"])
+		d.Set("team_id", data["team_id"])
```

Comment:
in this block starting around line 123 we define `data` to be a `map[string]interface{}` with only a `lease_id` key, so all of these `d.Set` calls would empty this values out, if we got here.

Reply:
Welp... my brain thought, for some reason that it was gonna do some kind of mutation on the data, not thinking the fact that it's a write call so it's providing the data... *sigh* My bad.